### PR TITLE
util: Removing JoinError in favor of fmt.Errorf

### PR DESCRIPTION
**Pull request by Thingee**
# Describe what this PR does #

Removing JoinError in favor of fmt.Errorf

## Is there anything that requires special attention ##

No

## Related issues ##

None

## Future concerns ##

List items that are not part of the PR and do not impact it's
functionality, but are work items that can be taken up subsequently.

**Checklist:**

* [x] **Commit Message Formatting**: Commit titles and messages follow
  guidelines in the [developer
  guide](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#commit-messages).
* [ ] Reviewed the developer guide on [Submitting a Pull
  Request](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#development-workflow)
* [x] [Pending release
  notes](https://github.com/ceph/ceph-csi/blob/devel/PendingReleaseNotes.md)
  updated with breaking and/or notable changes for the next major release.
* [x] Documentation has been updated, if necessary.
* [x] Unit tests have been added, if necessary.
* [x] Integration tests have been added, if necessary.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

* `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)

</details>

### CODE REVIEW

File: internal/cephfs/core/volume.go [hunk 1]
```diff
@@ -303,10 +303,10 @@ func (s *subVolumeClient) PurgeVolume(ctx context.Context, force bool) error {
 	if err != nil {
 		log.ErrorLog(ctx, "failed to purge subvolume %s in fs %s: %s", s.VolID, s.FsName, err)
 		if strings.Contains(err.Error(), cerrors.VolumeNotEmpty) {
-			return util.JoinErrors(cerrors.ErrVolumeHasSnapshots, err)
+			return fmt.Errorf("Failed as %w (internal %w", cerrors.ErrVolumeHasSnapshots, err)
```
Comment (Madhu-1) [inline internal/cephfs/core/volume.go]: missing closing bracket for internal

File: internal/util/errors.go [hunk 2]
```diff
@@ -57,9 +57,3 @@ func (e pairError) Is(target error) bool {
 func (e pairError) Unwrap() error {
 	return e.second
 }
-
-// JoinErrors combines two errors. Of the returned error, Is() follows the first
-// branch, Unwrap() follows the second branch.
-func JoinErrors(e1, e2 error) error {
```
Comment (Madhu-1) [inline internal/util/errors.go:63]: I believe that we can remove the pairError struct and all of its methods.

**Comment by Madhu-1** [comment]
https://github.com/ceph/ceph-csi/pull/4622 should fix the CI problem.

File: internal/util/errors_test.go [hunk 3]
```diff
@@ -30,42 +30,3 @@ var (
 func wrapError(e error) error {
 	return fmt.Errorf("w{%w}", e)
 }
-
```
Comment (Madhu-1) [inline internal/util/errors_test.go:33]: @Thingee changes looks good to be, we can remove this file as we don't need `wrapError` not `errFoo` and `errBar` anymore as they are not used.

**Review by Madhu-1 — COMMENTED** [review]
@Thingee can you please rebase this PR on top of devel branch to pick new CI changes?

**Review by Madhu-1 — APPROVED** [review]

**Review by Rakshith-R — APPROVED** [review]

**Comment by Rakshith-R** [comment]
@mergifyio queue

**Comment by Rakshith-R** [comment]
@mergifyio rebase

**Comment by Rakshith-R** [comment]
@mergifyio queue

**Pull request merged**